New benchmark: CaselawQA

RicardoDominguez opened the pull request:
Hi,

New task contribution for the CaselawQA benchmark for legal text annotation, introduced in the [Lawma: The Power of Specialization for Legal Tasks](https://arxiv.org/abs/2407.16615) paper appearing at ICLR2025.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/EleutherAI/lm-evaluation-harness?pullRequest=2739) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/EleutherAI/lm-evaluation-harness?pullRequest=2739) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/EleutherAI/lm-evaluation-harness?pullRequest=2739) it.</sub>

StellaAthena commented:
Thank you for the PR! Your paper doesn't seem to specify the evaluation implementation used in the paper. Is this the official implementation? If not, have you validated that models behave the same on this as they do in the implementation used in the paper?

RicardoDominguez commented:
Yes, this is the official implementation.

StellaAthena commented:
Great. Can you run the precommit hook to address the failing tests?

baberabb commented on lm_eval/tasks/caselawqa/caselawqa_sc.yaml:
```diff
@@ -0,0 +1,3 @@
+include: _default_template.yaml
```
also could you run the pre-commit
```
pip install pre-commit
pre-commit install
pre-commit run --all-files
```

RicardoDominguez replied:
done!

RicardoDominguez commented:
Changes:
* Removed .yaml extension from the default templates
* Renamed `_tiny` subtask to `_1k` which is more clear
* Made CoT evaluation the dafault, rather than MMLU-style directQA, since newer model >= 3B params perform better with CoT in this benchmark
* Run the precommit hook

RicardoDominguez commented:
One question: our benchmark aggregates over 260 different annotation problems. While it provides a measure of overall model performance, researchers might want to evaluate models' accuracy on specific annotation problems, say `caselawqa_sc_adminaction`. I've implemented these tasks on a different branch, see [here](https://github.com/socialfoundations/lm-evaluation-harness/tree/caselawqa_subtasks/lm_eval/tasks/caselawqa-substasks). In practice, these tasks only differ by their `dataset_name`. However, it might be unreasonable to "spam" the lm_eval library by implementing each of these 260 sets of annotation problems as a distinct task. Thoughts?

StellaAthena commented:
@baberabb Can you advise on subtask implementation? I'm not sure if we currently support it, but one thing that comes to mind is that it could be helpful to be able to algorithmically iterate over subtasks and report subtask score when requested by a user.

baberabb commented:
Right now we do expect each individual sub-task to have its own config, but it's worth thinking about having the option to create them programmatically at runtime, esp. if they all share the same base config with a minor differences. The number of configs are getting a bit unwieldy and we need to parse them all at startup.

@RicardoDominguez Can you add a link to your branch in the readme? I think that would be helpful for those users who want to make use of it.

Also if you could add a entry to `lm_eval/tasks/README.md` describing your benchmark in a sentence like all the other tasks!
